### PR TITLE
In development release notes category

### PR DESCRIPTION
**Pull request by ouchadam**
Adds a new release notes category for _in development_ code changes, this will help ensure all code changes can be related back to a release even if they're not user facing at the time of merge. 

- Includes recently merged _in development_ PRs

<details>

<summary> Draft output</summary>

Changes in Element 1 (2022-01-14)
=================================

Features ✨
----------
 - Enabling native support for window resizing ([#4811](https://github.com/vector-im/element-android/issues/4811))

Bugfixes 🐛
----------
 - Fix for stuck local event messages at the bottom of the screen ([#516](https://github.com/vector-im/element-android/issues/516))
 - Explore Rooms overflow menu - content update include "Create room" ([#3932](https://github.com/vector-im/element-android/issues/3932))
 - Fix a wrong network error issue in the Legals screen ([#4935](https://github.com/vector-im/element-android/issues/4935))

In development 🚧
--------------
 - Disabling onboarding automatic carousel transitions on user interaction ([#4914](https://github.com/vector-im/element-android/issues/4914))
 - Locking phones to portrait during the FTUE onboarding ([#4918](https://github.com/vector-im/element-android/issues/4918))
 - Adds a messaging use case screen to the FTUE onboarding ([#4927](https://github.com/vector-im/element-android/issues/4927))

SDK API changes ⚠️
------------------
 - `StateService.sendStateEvent()` now takes a non-nullable String for the parameter `stateKey`. If null was used, just now use an empty string. ([#4895](https://github.com/vector-im/element-android/issues/4895))

Other changes
-------------
 - Fix integration tests and add a comment with results (still not perfect due to github actions resource limitations) ([#4842](https://github.com/vector-im/element-android/issues/4842))
 - "/kick" command is replaced with "/remove". Also replaced all occurrences in string resources ([#4865](https://github.com/vector-im/element-android/issues/4865))
 - Add signing config for the release buildType. No secret added ([#4926](https://github.com/vector-im/element-android/issues/4926))

</details>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -61,6 +61,7 @@ Supported filename extensions are:
 
 - ``.feature``: Signifying a new feature in Element Android or in the Matrix SDK.
 - ``.bugfix``: Signifying a bug fix.
+- ``.dev``: Signifying a change which is not ready to be user facing, typically a component of a larger feature.
```
Comment (ouchadam) [inline CONTRIBUTING.md:64]: I'm not sure if `.dev` is the best extension might be a little too generic, what do people think?

Comment (bmarty) [reply CONTRIBUTING.md:64]: Maybe `.wip` since it's just an internal choose?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -61,6 +61,7 @@ Supported filename extensions are:
...
 - ``.doc``: Signifying a documentation improvement.
 - ``.removal``: Signifying a deprecation or removal of public API. Can be used to notifying about API change in the Matrix SDK
```
Comment (bmarty) [inline CONTRIBUTING.md:66]: Maybe take the opportunity to create a `.sdk` extension, this one is not obvious at first sight.

File: CONTRIBUTING.md (hunk 1 above)
Comment (ouchadam) [reply CONTRIBUTING.md:64]: sounds good to me :+1:

File: CONTRIBUTING.md (hunk 2 above)
Comment (ouchadam) [reply CONTRIBUTING.md:66]: do we want to replace `.removal` with `.sdk`?

Comment (bmarty) [reply CONTRIBUTING.md:66]: `.removal` is available out of the box, so I decided to hack it and use it to log the SDK change (especially in the API). Maybe just add `.sdk` and start using it is enough. Also need to update https://github.com/vector-im/element-android/blob/main/towncrier.toml#L23

Comment (ouchadam) [reply CONTRIBUTING.md]: if I understood correctly https://github.com/vector-im/element-android/pull/4953/commits/7259324b3ba1d871ba389a8239a5acc592e22110

if we want to keep the `removal` category is there a preferred section title?

Comment (bmarty) [reply CONTRIBUTING.md]: I think we will stop using `.removal`

**Review by bmarty — APPROVED** [review]

**Pull request merged**